### PR TITLE
feat: include node configurations

### PR DESCRIPTION
**Pull request by varnastadeus**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,44 +1,64 @@
-resource "castai_eks_cluster" "my_castai_cluster" {
-  account_id                 = var.aws_account_id
-  region                     = var.aws_cluster_region
-  name                       = var.aws_cluster_name
-  subnets                    = var.subnets
-  override_security_groups   = var.override_security_groups
-  dns_cluster_ip             = var.dns_cluster_ip
-  ssh_public_key             = var.ssh_public_key
-  tags                       = var.tags
+resource "castai_eks_cluster" "this" {
```
Comment (aldor007) [inline main.tf:1]: I like this rename but when updating terraform module user will remove and re-onboard the cluster. We should warn about it.

File: variables.tf [hunk 2]
```diff
@@ -103,7 +68,18 @@ variable "agent_aws_secret_access_key" {
...
+
+variable "node_configurations" {
+  type        = any
+  description = "Map of EKS node configurations to create"
+  default     = {}
+}
+
+variable "default_node_configuration" {
+  type        = string
+  description = "ID of the default node configuration"
```
Comment (aldor007) [inline variables.tf:84]: nitpick: IMO this variable should have a default - default node configuration for a cluster

File: variables.tf [hunk 3]
```diff
@@ -103,7 +68,18 @@ variable "agent_aws_secret_access_key" {
 }
 
 variable "castai_components_labels" {
-  type = map
+  type        = map
```
Comment (aldor007) [inline variables.tf:71]: nitpick: Why ąnot object?
```
  type     = map(object({
  disk_cpu_ratio = any
  subnets = any
  })
```

Using an object will serve as documentation of what fields are available to use

File: variables.tf (hunk 2 above)
Comment (varnastadeus) [reply variables.tf:84]: but if cluster is onboarded with node config, then we know node config id only when it is created

thats why default is set like this:

```hcl
default_node_configuration = module.cast-eks-cluster.castai_node_configurations["default"]
```

File: main.tf (hunk 1 above)
Comment (varnastadeus) [reply main.tf:1]: I will double check it but I don't think it has impact on actual resources

**Review by aldor007 — APPROVED** [review]

**Pull request merged**